lib/prism/pattern.rb

```diff
@@ -174,7 +182,13 @@ def compile_hash_pattern_node(node)
 
       preprocessed =
         node.elements.to_h do |element|
-          [element.key.unescaped.to_sym, compile_node(element.value)]
+          key = element.key
+          if key.respond_to?(:unescaped)
```

Comment:
`element` here is an `AssocNode`. Its `key` property is typed as just any node according to config.yml, and since not all nodes implement `unescaped`, I needed this check.

Should I revert and use a steep:ignore? Or if you know what all the possible key types could be, I could adjust config.yml.

Reply:
Ahh I see. Hash patterns can only have `SymbolNode` keys through their `AssocNode` nodes, but that's implicit. I think I'm fine with this line if you change it to:

```suggestion
          if key.is_a?(SymbolNode)
```

I'm assuming steep is smart enough to then not need the type annotation on the next line?

Reply:
Yep. Thanks for confirming. Changing to a `is_a?(SymbolNode)` and no longer need the type annotation as well.

Reply:
Addressed in 2d859af588500f4a708f6415feeab6c56b85bcf8